src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -633,6 +624,24 @@ def process_declarations(self, parent, nodes, arg_list):
                                 "Could not process {0}. Unrecognized "
                                 "attribute '{1}'.".format(decl.items,
                                                           str(attr)))
+                    elif isinstance(attr, Fortran2003.Intent_Attr_Spec):
+                        (_, intent) = attr.items
+                        normalized_string = \
+                            intent.string.lower().replace(' ', '')
+                        if normalized_string == "in":
+                            interface = ArgumentInterface(
+                                ArgumentInterface.Access.READ)
+                        elif normalized_string == "out":
+                            interface = ArgumentInterface(
+                                ArgumentInterface.Access.WRITE)
+                        elif normalized_string == "inout":
+                            interface = ArgumentInterface(
+                                ArgumentInterface.Access.READWRITE)
+                        else:
+                            raise InternalError(
```

Comment:
Please update the docstring to describe this error

Reply:
Added docstring